tests/utils.py

```diff
@@ -95,6 +98,35 @@ def run_fawltydeps(
     return proc.stdout.strip(), proc.stderr.strip(), proc.returncode
 
 
+def run_fawltydeps_function(
+    *args: str,
+    config_file: Path = Path("/dev/null"),
+    to_stdin: Optional[str] = None,
+    cwd: Optional[Path] = None,
```

Comment:
I understand the impulse to reuse the same API as for `run_fawltydeps()`, but if we're not actually going to change the current working directory here (see my `monkeypatch.chdir` suggestion above), then this should be renamed so as to not mislead the caller:
```suggestion
    basepath: Optional[Path] = None,
```